openff/toolkit/topology/topology.py

```diff
@@ -2385,3 +2386,25 @@ def hierarchy_iterator(
             if hasattr(molecule, iter_name):
                 for item in getattr(molecule, iter_name):
                     yield item
+
+    def __getattr__(self, name: str) -> List["HierarchyElement"]:
+        """If a requested attribute is not found, check the hierarchy schemes"""
+        # Avoid attempting to process dunder methods as hierarchy scheme iterator names
+        if name.startswith("__"):
+            raise AttributeError
+
+        if name in dir(self):
+            raise HierarchyIteratorNameConflictError(
+                f"Name {name} is already defined as an attribute/method of `Topology`."
+            )
+
+        try:
+            return [
+                element
+                for molecule in self.molecules
+                for element in getattr(molecule, name)
+            ]
+        except AttributeError as error:
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute {name!r}"
+            ) from error
```

Comment:
(not blocking) I think the nested error here is a pretty good idea. It may be good to use this chance to make the error message more informative instead of mirroring the Python default AttributeError. Maybe something like "Before raising this AttributeError, the Topology class checked whether the Molecules it contains have a HierarchyScheme with a matching name, however at least one Molecule in this Topology does not."